src/gazebo_mavlink_interface.cpp

```diff
@@ -392,7 +392,17 @@ void GazeboMavlinkInterface::Load(physics::ModelPtr _model, sdf::ElementPtr _sdf
 
     presetManager->GetCurrentProfileParam("max_step_size", param);
     const double max_step_size = our_any_cast<double>(param);
-    if (1.0 / real_time_update_rate != max_step_size)
+
+    // Adapt the real_time_update_rate according to the speed
+    // that we ask for in the env variable.
+    // Makes sure we don't do this multiple times.
+    if (1.0 / (real_time_update_rate / speed_factor_) != max_step_size) {
+      real_time_update_rate *= speed_factor_;
```

Comment:
I think this is the real place for the blame. Shouldn't it be expected that the `speed_factor_` is always expected to be the speed_factor_ compared to realtime and not whatever the simulation is set? Could we just do something like `real_time_update_rate = speed_factor_` ?

Reply:
Yeah, that does make sense. I am currently away from my workstation, but once I get back in 6 or so hours I can try and recompile with setting it absolutely rather than multiplying relatively.

Something confusing: whenever I was testing this a few weeks back, the speed factor shown on the bottom of Gazebo didn't change *unless* I multiplied `real_time_update_rate`, which is usually something like `0.250`. It appears as if the math for the update rate is much less based on the `speed_factor_`, and based more upon the change that happens to `real_time_update_rate` (i.e. if it is 2x normal, 4x normal, etc.)